Option to disable keychain credentials

sagerb opened the pull request:
<!-- Provide a general summary of your changes in the title. -->
<!-- Examples: Updates pull request template -->

## Intent

<!-- Describe what problem you are addressing in this pull request. -->
<!-- If this change is associated with an open issue, please link to it here. -->
<!-- Example: "Resolves #24" -->
<!-- See https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue -->

This PR introduces a configuration option to disable the use of the KeyRing for credential storage and instead use the file system. This is extremely useful for development as well as a good escape hatch for customers who may not have a working KeyRing environment (or preference there-towards).

Resolves #2519 

## Type of Change

<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
<!-- If you check more than one box, you may need to refactor this change into separate pull requests -->

- - [ ] Bug Fix <!-- A change which fixes an existing issue -->
- - [x] New Feature <!-- A change which adds additional functionality -->
- - [ ] Breaking Change <!-- A breaking change which causes existing functionality to change -->
- - [ ] Documentation <!-- User or developer documentation -->
- - [ ] Refactor <!-- Code restructuring -->
- - [ ] Tooling <!-- Build, CI, or release scripts and configuration -->

## Approach

<!-- Describe how you solved this problem and any trade-offs you encountered. -->
<!-- Link any additional documentation associated with this change here -->

I first added a new configuration option for the publisher extension, which is enabled by default:
![2025-01-31 at 11 23 AM](https://github.com/user-attachments/assets/9c083eb0-e5ad-47e2-91b3-1568e1543ee8)

I then added a corresponding command line option for the agent. This was a more straightforward change than passing the option for every credential-related call. The impact of this decision is that the user must restart the extension after changing the configuration setting. Since this won't be very common, I believe this was a reasonable optimization to keep the complexity low.
```
$(just executable-path) ui -h
Usage: publisher ui [<path>] [flags]

Arguments:
  [<path>]    Sets the current working directory for the agent.

Flags:
  -h, --help                  Show context-sensitive help.
  -v, --verbose               Enable verbose logging. Use -vv or --verbose=2 for debug logging.

      --listen=HOST[:PORT]    Network address to listen on.
      --use-keychain          Use Keychain services to store/manage credentials.
```

After those two changes were in place, I added a helper function within the extension to access the setting, then updated the service spawning code to reflect the current configuration value.  The agent then stores that value and each time the agent's credential code initializes, it either bypasses or attempts to use the KeyChain dependency.

The agent's debug launch profile has been updated to support launching the agent with the command line set. I set the initial value to `false` as I found myself wanting to debug using the file-based credentials more often than the KeyChain alternative. If this is not desired by the developer, they can easily change it within their own config.

## User Impact

<!-- Describe how this change will affect users. -->
<!-- Specifically changes to the flow and trade-offs we are making. -->

New feature, but default maintains the behavior they had previously.

## Automated Tests

<!-- Describe the automated tests associated with this change. -->
<!-- If automated tests are not included in this change, please state why. -->

Unit tests added to validate the desired behavior.

## Directions for Reviewers

<!-- Provide steps for reviewers to validate this change manually. -->

Launching a debug version of the extension from your VSCode instance for the extension, will work for verification. The configuration value is respected and maintained after being set, when you then reload the window with `Developer: Reload Window`. I'd suggest verification of the functionality by:
1. Confirm the configuration setting for the extension is set to enabled (checked). If have to change it, then reload the VSCode window
2. Activate the publisher extension and view the credentials. You should be able to confirm that your existing credentials is listed.
3. Change the configuration setting to disabled and reload the window. 
4. Activate the publisher extension and view the credentials. You should either have none or whatever you had stored within the `~.connect-credentials` file.
5. Add a new credential and make sure it shows up in your credential list
6. Reload the window again and make sure the credentials have remained.
7. Change the configuration setting back to enabled and reload the window
8. You should be able to validate that the original KeyChain based credentials are now shown.


## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply: -->
<!--- If you need clarification on any of these, feel free to ask. We're here to help! -->

- [ ] I have updated [CHANGELOG.md](../CHANGELOG.md) to cover notable changes.

dotNomad reviewed (commented):
It looks like this was branched off of the CLI removal. Can we change the base, or rebase this to isolate just the keychain credential changes? That will probably get CI passing again.

sagerb commented:
> It looks like this was branched off of the CLI removal. Can we change the base, or rebase this to isolate just the keychain credential changes? That will probably get CI passing again.

Oh yes, sorry about that. I've rebased the diff and updated from that updated branch as well. Thanks!

dotNomad reviewed (approved):
This is very clean. I just had a suggestion to trim up the description of the setting, and make it more similar to other VS Code setting descriptions.

sagerb commented:
To get CI to pass, we have to merge #2584 first.
